src/main/java/com/project/springbootproject/service/user/UserServiceImpl.java

```diff
@@ -0,0 +1,33 @@
+package com.project.springbootproject.service.user;
+
+import com.project.springbootproject.dto.userdto.UserDto;
+import com.project.springbootproject.dto.userdto.UserRequestDto;
+import com.project.springbootproject.exception.RegistrationException;
+import com.project.springbootproject.mapper.UserMapper;
+import com.project.springbootproject.model.User;
+import com.project.springbootproject.repository.user.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserDto register(UserRequestDto userRequestDto) {
+        if (userRepository.findByEmail(userRequestDto.getEmail()).isPresent()) {
+            throw new RegistrationException("Can not register user");
+        }
+
+        User user = new User();
+        user.setEmail(userRequestDto.getEmail());
+        user.setPassword(userRequestDto.getPassword());
+        user.setFirstName(userRequestDto.getFirstName());
+        user.setLastName(userRequestDto.getLastName());
+        user.setShippingAddress(userRequestDto.getShippingAddress());
```

Comment:
still not sure why you hesitate to use mapper
```suggestion
        User user = userMapper.toUserModel(userRequestDto);
```

Reply:
https://mate.academy/learn/java-spring-boot/java-spring-boot-security-basics?section=video&videoId=6179 
In this video, Bohdan advises to use this approach, but he didnt explain it